pkg/client/git/client.go

```diff
@@ -82,10 +82,12 @@ func (c *ClientFactory) getTokenFromSecret(secretRef *v1.SecretReference) (token
 	}
 
 	switch gitSecret.Type {
-	case v1.SecretTypeBasicAuth:
+	case v1.SecretTypeBasicAuth, "credential.devops.kubesphere.io/basic-auth":
```

Comment:
It would be better if we use 
https://github.com/kubesphere/ks-devops/blob/69dda0b28cad59fe187c5d9bce972fee3ebfb7f8/pkg/api/devops/v1alpha3/credential_types.go#L34

Reply:
IMO, it's not ok to let a git client use the package `ks-devops/pkg/api/devops/v1alpha3`